BullsAndCows/SecretGenerator.cs

```diff
@@ -6,7 +6,24 @@ public class SecretGenerator
     {
         public virtual string GenerateSecret()
         {
-            throw new NotImplementedException();
+            Random ran = new Random();
+            string secret = ran.Next(0, 10).ToString();
+            ConstructNumber();
+            ConstructNumber();
+            ConstructNumber();
+
+            void ConstructNumber()
+            {
+                string nextNumber = ran.Next(0, 10).ToString();
+                while (secret.Contains(nextNumber))
```

Comment:
While loop might be a little bit dangerous. Try refactor to pop from a finite list.